guardian/prosemirror-elements
# feat: Add interactive element

## What does this change?

Building on @rhystmills' excellent work in #65, adds an interactive element:

<img width="832" alt="Screenshot 2021-12-07 at 15 26 59" src="https://user-images.githubusercontent.com/7767575/145058634-ec9c7f0c-1804-4ecd-b61a-6f135eb469fb.png">

I've added a link to the original interactive source, which wasn't present in the original.

I've also added conditional display of the iframeUrl via our `iframeUrl` parameter. If the source data contains an iframe, we display it – else we fall back to the interactive HTML, or display nothing.

This also includes some improvements to the Preview height calculations.

## How to test

Add this element directly in the demo page. It should behave as expected. Unfortunately, our ability to modify some of the fields in question is limited to code changes – in the absence of a decent UI for this, perhaps try altering some fields in the code and instantiate new instances.

## How can we measure success?

A shiny new element that matches (or improves) our current experience when working with interactives.

## Have we considered potential risks?

We'll  need to test thoroughly in our consuming CMS before shipping. 

## Accessibility
